stratum/hal/lib/phal/phal_cli.cc

```diff
@@ -0,0 +1,457 @@
+        if (sscanf(val_str.c_str(), "%ld", &val) != 1) {
+          std::cout << "Invalid int64" << std::endl;
+          continue;
+        }
+        update->mutable_value()->set_int64_val(val);
+        break;
+
+      } else if (type.compare("uint32") == 0) {
+        uint32_t val;
+        if (sscanf(val_str.c_str(), "%d", &val) != 1) {
+          std::cout << "Invalid uint32" << std::endl;
+          continue;
+        }
+        update->mutable_value()->set_uint32_val(val);
+        break;
+
+      } else if (type.compare("uint64") == 0) {
+        uint64_t val;
+        if (sscanf(val_str.c_str(), "%ld", &val) != 1) {
+          std::cout << "Invalid uint64" << std::endl;
+          continue;
+        }
+        update->mutable_value()->set_uint64_val(val);
+        break;
+
+      } else if (type.compare("double") == 0) {
+        double val;
+        if (sscanf(val_str.c_str(), "%lf", &val) != 1) {
+          std::cout << "Invalid double" << std::endl;
+          continue;
+        }
+        update->mutable_value()->set_double_val(val);
+        break;
+
+      } else if (type.compare("float") == 0) {
+        float val;
+        if (sscanf(val_str.c_str(), "%f", &val) != 1) {
+          std::cout << "Invalid float" << std::endl;
+          continue;
+        }
+        update->mutable_value()->set_float_val(val);
+        break;
+
+      } else if (type.compare("bool") == 0) {
+        if (val_str.compare("false") == 0) {
+          update->mutable_value()->set_bool_val(false);
+          break;
+
+        } else if (val_str.compare("true") == 0) {
+          update->mutable_value()->set_bool_val(true);
+          break;
+
+        } else {
+          std::cout << "Invalid bool <false|true>" << std::endl;
+          continue;
+        }
+
+      } else if (type.compare("string") == 0) {
+        update->mutable_value()->set_string_val(val_str);
+        break;
+
+      } else if (type.compare("bytes") == 0) {
+        update->mutable_value()->set_bytes_val(val_str);
+
+      } else {
+        std::cout << "Must specify a type: <int32, uint32, int64, "
+                  << "uint64, double, float, bool, string, bytes" << std::endl;
+        continue;
+      }
+    }
+
+    absl::Time start_time = absl::Now();
+    auto status = phaldb_svc_->Set(&context, req, &resp);
+    absl::Time execute_time = absl::Now();
+
+    // RPC failed
+    if (!status.ok()) {
+      return MAKE_ERROR(ERR_INTERNAL)
+             << "gRPC Get call failed: " << status.error_message();
+    }
+
+    int64_t execute_duration =
+        (execute_time - start_time) / absl::Microseconds(1);
+
+    auto result_str = resp.DebugString();
+    if (result_str.size() <= 0) {
+      std::cout << "No Results" << std::endl;
+    } else {
+      std::cout << result_str << std::endl;
+    }
+    std::cout << "Executed query in " << execute_duration << " us."
+              << std::endl;
+    return ::util::OkStatus();
+  }
+
+  enum cmd_type { Get, Subscribe, Set };
```

Comment:
I'd suggest using `CMD_GET`, `CMD_SUBSCRIBE`, and `CMD_SET`. 
https://google.github.io/styleguide/cppguide.html#Enumerator_Names